### DISCUSSION TITLE
how to set/get Cookie from WebSocket?

### DISCUSSION THREAD

**Discussion by Alnyz** [fastapi/fastapi#10658] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import (
    Cookie,
    Depends,
    FastAPI,
    Request,
    Form,
    WebSocket,
    WebSocketException,
    status,
)

async def get_cookie_or_token(
    websocket: WebSocket,
    session: str | None = Cookie(default=None),
): 
    if session is None:
        raise WebSocketException(code=status.WS_1008_POLICY_VIOLATION)
    return session

@app.get("/chat")
@app.post("/chat")
def chat(request: Request, _i: str | None = Cookie(default=None)):
    name = request.cookies.get("name")
    return templates.TemplateResponse("chat.html", context={
        "request": request,
        "name": name,
    })

@app.post("/")
def home_login(request: Request, username: str = Form(...)):
    uid = gen_uuid()
    data = {
            uid: {
                'uname': username
            }
        }
    manager.active_connections.update(
        data
    )
    res = RedirectResponse("/chat")
    cookie_data = encrypt_jwt(data)
    res.set_cookie("_i", cookie_data)
    return res

@app.websocket("/ws")
async def websocket_endpoint(websocket: WebSocket,
                            _i: str | None = Cookie(default=None)):
    print(_i)
    await manager.connect(websocket)
```

### Description

at `websocket_endpoint` function i have try to `print(_i)` which is cookie i've declare from path `/` but it return `None` also when i check to `network` tab there is no cookies. 
also i have try to use `cookie: Annotated[str, Depends(get_cookie_or_token)]` as parameter in `websocket_endpoint` function

### Operating System

Windows

### FastAPI Version

0.104.0

### Pydantic Version

1.10.13

### Python Version

3.11.5

**Comment by hasansezertasan** [comment, marked as answer]
Can you please try `websocket.cookies.get("name")`?
